js/src/common/components/Modal.js

```diff
@@ -35,7 +35,8 @@ export default class Modal extends Component {
       this.attrs.animateHide();
       // Here, we ensure that the animation has time to complete.
       // See https://mithril.js.org/lifecycle-methods.html#onbeforeremove
-      return new Promise((resolve) => setTimeout(resolve, 1000));
+      // Bootstrap's default animation duration is 300 ms.
```

Comment:
One last suggestion (I hope!), I'd suggest the following comment so that one can more easily find where the 300ms is defined in bootstrap's source code, should it be necessary.

```suggestion
      // Bootstrap's Modal.TRANSITION_DURATION is 300 ms.
```